cmd/buildctl/build.go

```diff
@@ -191,6 +191,11 @@ func buildAction(clicontext *cli.Context) error {
 	} else {
 		exports, err = build.ParseOutput(clicontext.StringSlice("output"))
 	}
+	defer func() {
```

Comment:
Why is this defer here where it would be called on parse errors without the build even running. Shouldn't it be inside the component that actually writes the file?

Reply:
Hi @tonistiigi 
The empty tarball at #1305 is created at L190/L192, so I think the defer should be put right after that. What's your opinion? 

The creater is in `package build`, do you mean that this cleanup handling should also called by `build`? Currently `build` doesn't saves the info of `exports`, so we should also add a struct in `buid` to reserve it? 🤔

Reply:
Ideally in https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/output.go#L112 we would rewrite the wrapper so that `os.Create` is called before first `Write()` call (we can check that file is writable before). Plus we can add `CloseWithError()` callback that may remove the file if error happens mid-write.

Reply:
Yep, rewrite the wrapper is better~

Your help is needed here, actually I don't know how to rewrite it :face_with_head_bandage: